src/Data/Singletons/Promote/Type.hs

```diff
@@ -29,10 +29,8 @@ promoteType = go []
     go args     (DAppT t1 t2) = do
       k2 <- go [] t2
       go (k2 : args) t1
-    go args     (DSigT ty _) = go args ty  -- just ignore signatures
-    go []       (DVarT name) = return $ DVarT name
-    go _        (DVarT name) = fail $ "Cannot promote an applied type variable " ++
-                                      show name ++ "."
+    go args     (DSigT ty ki) = DSigT <$> go args ty <*> go [] ki
```

Comment:
To be honest, I didn't entirely understand what I was doing here. I'll add a test.